pixeltable/io/label_studio.py

```diff
@@ -0,0 +1,612 @@
+        tasks_to_delete = [existing_tasks[rowid]['id'] for rowid in deleted_rowids]
+
+        if len(tasks_to_delete) > 0:
+            self.project.delete_tasks(tasks_to_delete)
+            print(f'Deleted {len(tasks_to_delete)} tasks(s) in {self} that are no longer present in Pixeltable.')
+
+        # Remove them from the `existing_tasks` dict so that future updates are applied correctly
+        for rowid in deleted_rowids:
+            del existing_tasks[rowid]
+
+        return SyncStatus(external_rows_deleted=len(deleted_rowids))
+
+    def __update_table_from_tasks(self, t: Table, tasks: dict[tuple, dict]) -> 'SyncStatus':
+        if ANNOTATIONS_COLUMN not in self.col_mapping.values():
+            return SyncStatus.empty()
+
+        local_annotations_column = next(k for k, v in self.col_mapping.items() if v == ANNOTATIONS_COLUMN)
+        annotations = {
+            # Replace [] by None to indicate no annotations. We do want to sync rows with no annotations,
+            # in order to properly handle the scenario where existing annotations have been deleted in
+            # Label Studio.
+            tuple(task['meta']['rowid']): task[ANNOTATIONS_COLUMN] if len(task[ANNOTATIONS_COLUMN]) > 0 else None
+            for task in tasks.values()
+        }
+
+        # Prune the annotations down to just the ones that have actually changed.
+        rows = t.select(t[local_annotations_column])
+        for row in rows._exec():
+            assert len(row.vals) == 1
+            if row.rowid in annotations and annotations[row.rowid] == row.vals[0]:
+                del annotations[row.rowid]
+
+        # Apply updates
+        updates = [{'_rowid': rowid, local_annotations_column: ann} for rowid, ann in annotations.items()]
+        if len(updates) > 0:
+            _logger.info(
+                f'Updating table `{t.get_name()}`, column `{local_annotations_column}` with {len(updates)} total annotations.'
+            )
+            # batch_update currently doesn't propagate from views to base tables. As a workaround, we call
+            # batch_update on the actual ancestor table that holds the annotations column.
+            # TODO(aaron-siegel): Simplify this once propagation is properly implemented in batch_update
+            ancestor = t
+            while local_annotations_column not in ancestor.tbl_version_path.tbl_version.cols_by_name:
+                assert ancestor.base is not None
+                ancestor = ancestor.base
+            ancestor.batch_update(updates)
+            print(f'Updated annotation(s) from {len(updates)} task(s) in {self}.')
+
+        return SyncStatus(pxt_rows_updated=len(updates))
+
+    def to_dict(self) -> dict[str, Any]:
+        return {
+            'name': self.name,
+            'project_id': self.project_id,
+            'media_import_method': self.media_import_method,
+            'col_mapping': self.col_mapping
```

Comment:
This will break when columns are renamed. You need to store qualified column ids instead (ie, table id/col id).